HomeWork13/HomeWork13/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿using static HomeWork13.First;
+
+namespace HomeWork13
+{
+    public class First
+    {
+        public delegate void FirstHandler(bool prob);
+
+        public static int Multiply(int x, int y) => x * y;
+    }
+
+    public class Second
+    {
+        private int _result;
+        public Predicate<int> Calc(int x, int y, Func<int, int, int> func)
+        {
+            _result = func(x, y);
+
+            Predicate<int> predicate = Result;
+
+            return predicate;
+        }
+
+        public bool Result(int x)
+        {
+            return _result % x == 0;
+        }
+    }
+    internal class Program
+    {
+        static void Show(bool result)
+        {
+            Console.WriteLine($"result: {result}");
+        }
+        static void Main(string[] args)
+        {
+            Second second = new();
+
+            var tig = second.Calc(10, 5, Multiply);
```

Comment:
I think we need to discuss this task as I don't understand it clearly

Reply:
Second second = new(); // Visual studio suggested ("new " expresion can be simplified) i want try this...

var tig = second.Calc(10, 5, Multiply); // tig - it's anonymous method Predicate<int> what return fron Calc method.
tig return bool and take 1 parametr for terms of comparison

Calc method from my task - 1) the “Calc” method, which takes a delegate for the “Multiply” method as input, as well as parameters for calculation. The "Calc" method makes a delegate call to the "Multiply" method and stores the result values inside the class. The return type of the "Calc" method is a delegate. This delegate must take 1 number as a parameter and return a bool value as a result. This delegate refers to the "Result" method.

(10, 5, Multiply) - 10 and 5 my own parametr - it's no metter what that be.

Reply:
> I think we need to discuss this task as I don't understand it clearly

Second second = new(); // Visual studio suggested ("new " expresion can be simplified) i want try this...

var tig = second.Calc(10, 5, Multiply); // tig - it's anonymous method Predicate what return fron Calc method.
tig return bool and take 1 parametr for terms of comparison

Calc method from my task - 1) the “Calc” method, which takes a delegate for the “Multiply” method as input, as well as parameters for calculation. The "Calc" method makes a delegate call to the "Multiply" method and stores the result values inside the class. The return type of the "Calc" method is a delegate. This delegate must take 1 number as a parameter and return a bool value as a result. This delegate refers to the "Result" method.

(10, 5, Multiply) - 10 and 5 my own parametr - it's no metter what that be.